tasks/src/main/scala/dagr/tasks/parallel/PathBasedScatterGatherPipeline.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+package dagr.tasks.parallel
+
+import java.nio.file.Path
+
+import dagr.core.tasksystem.{Linker, Task}
+import dagr.tasks.misc.DeleteFiles
+
+object PathBasedScatterGatherPipeline {
+
+  /** All tasks that split inputs should extend this trait and implement the method to specify the outputs.  The latter
+    * method will only be called once this task has completed successfully.
+    */
+  trait SplitInputPathTask extends Task {
+    def outputs: Iterable[Path]
+  }
+
+  /** All tasks that process intermediates should extend this trait and implement the method to specify the output.
+    * The latter method will only be called once this task has completed successfully.
+    */
+  trait ScatterPathTask extends Task {
+    def output: Path
+  }
+
+  /** All tasks that gather multiple outputs into a single output should implement this task. */
+  trait GatherPathTask extends Task
+
+  /** A wrapper for the split input task. */
```

Comment:
The idea of these traits was to have a more intuitive name for the underlying paths than `subDomains` or the like.